Feature integration: UI Revamp on Marketing Page

ecgan opened the pull request:
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR contains the whole functionality for the epic issue https://github.com/woocommerce/woocommerce/issues/33895. It merges the main feature branch into the trunk branch. It consists of the following subfeature PRs that have been reviewed and merged:

- https://github.com/woocommerce/woocommerce/pull/33964
- https://github.com/woocommerce/woocommerce/pull/34080
- https://github.com/woocommerce/woocommerce/pull/34264
- https://github.com/woocommerce/woocommerce/pull/34415
    - https://github.com/woocommerce/woocommerce/pull/34444

I am hoping to get this merged and make it into WooCommerce 7.0 release, if that is possible. (Code freeze for WC 7.0 is on Thursday 15 September 2022).

I will come up with more tests (playwright e2e tests and unit tests) in separate PR later. The test PR would be targeting trunk branch directly after this PR.

### How to test the changes in this Pull Request:

The following is a brief test. For more detailed tests instructions with screenshots and demo videos, please visit the subfeature PRs above.

1. Go to WooCommerce Settings > Advanced > Features, and enable the multichannel marketing feature.
2. Go to the Marketing page at `/wp-admin/admin.php?page=wc-admin&path=%2Fmarketing`. You should see the new beta Multichannel Marketing page UI.
3. In the page, you should see the "Discover more marketing tools" card. It should be collapsed by default. Expand the card and you should see a list of marketing extensions. 
4. You should see the "Learn about marketing a store" card. It should be collapsed by default.
5. Install a marketing extension, and come back to the Marketing page. 
6. You should see the "Installed extensions" card. The card should be expanded by default. 
7. Go back to WooCommerce Settings > Advanced > Features, and disable the multichannel marketing feature.
8. Go to the Marketing page at `/wp-admin/admin.php?page=wc-admin&path=%2Fmarketing`. You should see the existing normal Marketing page UI.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you successfully run tests with your changes locally?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm changelog add --filter=<project>`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

ecgan committed:
Add a feature toggle to switch between the old and new Marketing page (#33964)

ecgan committed:
Merge branch 'trunk' into feature/33895-marketing-page/main

ecgan committed:
Merge branch 'trunk' into feature/33895-marketing-page/main

Conflicts:
	pnpm-lock.yaml

ecgan committed:
Merge branch 'trunk' into feature/33895-marketing-page/main

ecgan committed:
Implement the "Installed extensions" card (#34080)

ecgan committed:
Merge branch 'trunk' into feature/33895-marketing-page/main

ecgan committed:
Marketing page - Implement "Learn about marketing a store" card (#34264)

ecgan committed:
Merge branch 'trunk' into feature/33895-marketing-page/main

ecgan committed:
Merge branch 'trunk' into feature/33895-marketing-page/main

ecgan committed:
Marketing page - Implement "Discover more marketing tools" card (#34415)

Co-authored-by: Nima <nima.karimi@automattic.com>
Co-authored-by: Nima Karimi <73110514+nima-karimi@users.noreply.github.com>

ecgan commented:
Hi @chihsuan , given that you have reviewed and tested some of the previous subfeature PRs (https://github.com/woocommerce/woocommerce/pull/34415 and https://github.com/woocommerce/woocommerce/pull/34444), I think you may be the best person to review this feature integration PR. Would you mind giving this PR a review please? 🙏

ecgan committed:
Add changelog.

ecgan committed:
Remove subfeature changelog.

chihsuan commented:
> Hi @chihsuan , given that you have reviewed and tested some of the previous subfeature PRs (#34415 and #34444), I think you may be the best person to review this feature integration PR. Would you mind giving this PR a review please? 🙏

Sure. Will review it today!

chihsuan commented on plugins/woocommerce/client/admin/config/core.json:
```diff
@@ -12,6 +12,7 @@
 		"shipping-setting-tour": true,
 		"homescreen": true,
 		"marketing": true,
+		"multichannel-marketing": false,
```
Hey @ecgan ~Do you plan to turn on this in a follow-up PR? I think we need to enable it before 7.0 code freeze.~ I guess we just want it to be off by default.

ecgan replied:
Yes, we want it to be off by default for now. This UI revamp is just the first step. There shall be more iterations adding more features to the Marketing page, and we shall turn it on by default when those features are done. More info is available here: pe2C5g-bI-p2.

chihsuan reviewed (dismissed):
Tested well and LGTM. 🚀

chihsuan commented:
Hey @ecgan Thanks for working on this. I know we intend to collapse the cards by default. But I've been curious why we do that. Also, why don't add `target="_blank"` for "Get Started" links? 🙂

ecgan commented:
@chihsuan ,

> I know we intend to collapse the cards by default. But I've been curious why we do that. 

That's because in future iterations, we will be showing more cards - "Channels" and "Campaigns" - above the collapsed cards. Those cards would be more important and they would be expanded by default.

More info is in the P2 project thread pe2C5g-bH-p2 and in figma design 7cNed5RvTasoLgn38Hea0R-fi-2116%3A81074.

> Also, why don't add `target="_blank"` for "Get Started" links? 🙂

Well I was just following the existing behavior in the "Recommended extensions" in the normal view (when we have the feature flag turned off; see screenshot below), when we click on the extensions here, they open in the same existing tab and not in a new tab. 😄 I am guessing it's because we want them to go to woocommerce.com, make the purchase and then use the "Add to site" functionality (see second screenshot below) which should then bring them back to their own WooCommerce store site. With this flow, there wouldn't be an "outdated" or "stale" browser tab that users need to reload manually.

<img width="730" alt="image" src="https://user-images.githubusercontent.com/417342/189876756-7949e884-916b-4e90-8511-dfe22abd4f9c.png">

<img width="514" alt="image" src="https://user-images.githubusercontent.com/417342/189877744-1110e8dc-b23d-4afe-b95d-3dee9d561a7c.png">

ecgan commented:
@chihsuan , thanks so much for reviewing this! 🙏 Again, I am not authorized to merge this PR into trunk. Could you or someone else help to merge this please?

ecgan committed:
Remove initialCollapsed for DiscoverTools.

This is so that we can have something meaningful displayed in the Marketing page, instead of just two collapsed cards.

ecgan commented:
> > I know we intend to collapse the cards by default. But I've been curious why we do that.
> 
> That's because in future iterations, we will be showing more cards - "Channels" and "Campaigns" - above the collapsed cards. Those cards would be more important and they would be expanded by default.

@chihsuan , on second thought, I think you raised an interesting point. When new users (who haven't installed any marketing extensions) visit the new beta Marketing page, they would see two collapsed cards, which isn't really meaningful. I have made a change in [`cf25f5d` (#34642)](https://github.com/woocommerce/woocommerce/pull/34642/commits/cf25f5d9961955bb4f2d787fa5cd6b6bbb083550) to make the DiscoverTools component expanded by default, and I can make it collapsed by default later when the time comes in future iterations.

chihsuan commented:
@ecgan Awesome. Thanks for explaining. 😃 

> on second thought, I think you raised an interesting point. When new users (who haven't installed any marketing extensions) visit the new beta Marketing page, they would see two collapsed cards, which isn't really meaningful. I have made a change in [cf25f5d (#34642)](https://github.com/woocommerce/woocommerce/pull/34642/commits/cf25f5d9961955bb4f2d787fa5cd6b6bbb083550) to make the DiscoverTools component expanded by default, and I can make it collapsed by default later when the time comes in future iterations.

That makes sense to me!

chihsuan commented:
> @chihsuan , thanks so much for reviewing this! 🙏 Again, I am not authorized to merge this PR into trunk. Could you or someone else help to merge this please?

@ecgan Looks like the required lint checks action failed. 😞  I think it's not related to your PR, but could you rebase this PR to see if that can fix it?

ecgan committed:
Merge branch 'trunk' into feature/33895-marketing-page/main

ecgan commented:
@chihsuan , I have merged trunk branch into this PR in [`53dbaff` (#34642)](https://github.com/woocommerce/woocommerce/pull/34642/commits/53dbaff4495fb45ac95b38b5a7ebc933c5daac18). The required lint checks action still failed (see https://github.com/woocommerce/woocommerce/actions/runs/3050662041/jobs/4917998900), with an error message `woocommerce/client/legacy:turbo:build: Fatal error: Unable to find local grunt.`. I manually re-run the job and then it works (see https://github.com/woocommerce/woocommerce/actions/runs/3050662041/jobs/4918201000). 

e2e tests was also failing (https://github.com/woocommerce/woocommerce/actions/runs/3050662043/jobs/4917999308), and I re-run the job, and then it works (https://github.com/woocommerce/woocommerce/actions/runs/3050662043/jobs/4918291597).

All the required checks are now OK. This PR is now ready for merging. 🙂
